src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -45,20 +46,43 @@ public static string HtmlEncode(string value)
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlEncode(value, index, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            var sb = new ValueStringBuilder(value.Length);
+            HtmlEncode(value, index, ref sb);
+            return sb.ToString();
         }
 
         public static void HtmlEncode(string value, TextWriter output)
         {
-            output.Write(HtmlEncode(value));
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+            if (string.IsNullOrEmpty(value))
+            {
+                output.Write(value);
+                return;
+            }
+
+            // Don't create StringBuilder if we don't have anything to encode
+            int index = IndexOfHtmlEncodingChars(value, 0);
+            if (index == -1)
+            {
+                output.Write(value);
+                return;
+            }
+
+            // a thought for possible future optimisation: rather than build the whole output string,
+            // build 4kb-sized chunks and dump them into the writer as you go
+            // Of course would only make a difference for large strings
```

Comment:
I expect in many cases the TextWriter that's passed in does its own buffering, such that if we were to skip the ValueStringBuilder and just issue writes to it, that's essentially what we'd be doing.

FWIW, I don't think we need this comment.  If this is something to be investigated, it'd be better as an issue in the repo.

Reply:
+1 it's also more likely that someone would do this investigation if it was moved to an issue.